src/extras/SceneUtils.js

```diff
@@ -37,9 +42,240 @@ var SceneUtils = {
 		scene.remove( child );
 		parent.add( child );
 
+	},
+
+	convertFromZUp: function ( root, animations ) {
+
+		// see https://gamedev.stackexchange.com/a/7932
+
+		// this matrix will change the coordinate system of an object hierarchy and
+		// animations (optional) from right-handed Z-up to right-handed Y-up
+
+		var matrix = new Matrix4().set(
+			1,   0,   0,   0,
+			0,   0,   1,   0,
+			0, - 1,   0,   0,
+			0,   0,   0,   1
+		);
+
+		var converter = new BasisConverter( matrix );
+		converter.convert( root, animations );
+
 	}
 
 };
 
+function BasisConverter( matrix ) {
+
+	this.matrix = matrix;
+
+}
+
+Object.assign( BasisConverter.prototype, {
+
+	convert: function ( root, animations ) {
+
+		var scope = this;
+
+		root.traverse( function( object ) {
+
+			// position, quaternion, scale
+
+			scope.convertMatrix4( object.matrix );
+			object.matrix.decompose( object.position, object.quaternion, object.scale );
+
+			// geometry
+
+			var geometry = object.geometry;
+
+			if ( geometry !== undefined &&  geometry.isBufferGeometry === true ) {
+
+				var position = geometry.attributes.position;
+				var normal = geometry.attributes.normal;
+
+				var matrix = scope.matrix;
+
+				if ( position !== null ) {
+
+					matrix.applyToBufferAttribute( position );
+
+				}
+
+				if ( normal !== null ) {
+
+					var normalMatrix = new Matrix3().setFromMatrix4( matrix );
```

Comment:
Either use a closure, or forget the normal matrix (as you did previously) and note that it is assumed there is no non-uniform scaling in the change-of-basis matrix. I'd do the latter, now that I think about it.